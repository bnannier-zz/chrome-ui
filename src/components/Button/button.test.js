import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

import { Button } from "../";

describe("Button Tests", () => {
  it("Default", function() {
    const wrapper = shallow(<Button>test</Button>);
    expect(wrapper).toMatchSnapshot();
  });

  it("Raised", function() {
    const wrapper = shallow(<Button raised>test</Button>);
    expect(wrapper).toMatchSnapshot();
  });

  it("Unelevated", function() {
    const wrapper = shallow(<Button unelevated>test</Button>);
    expect(wrapper).toMatchSnapshot();
  });

  it("Outlined", function() {
    const wrapper = shallow(<Button outlined>test</Button>);
    expect(wrapper).toMatchSnapshot();
  });

  it("Dense", function() {
    const wrapper = shallow(<Button dense>test</Button>);
    expect(wrapper).toMatchSnapshot();
  });

  it("Icon", function() {
    const wrapper = shallow(<Button icon>test</Button>);
    expect(wrapper).toMatchSnapshot();
  });
});
