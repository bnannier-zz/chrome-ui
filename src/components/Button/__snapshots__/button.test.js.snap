// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Tests Default 1`] = `
<Base>
  <button
    className={
      Array [
        "mdc-button",
        Object {
          "material-icons": undefined,
          "mdc-button--dense": undefined,
          "mdc-button--outlined": undefined,
          "mdc-button--raised": undefined,
          "mdc-button--unelevated": undefined,
        },
      ]
    }
  >
    test
  </button>
</Base>
`;

exports[`Button Tests Dense 1`] = `
<Base
  dense={true}
>
  <button
    className={
      Array [
        "mdc-button",
        Object {
          "material-icons": undefined,
          "mdc-button--dense": true,
          "mdc-button--outlined": undefined,
          "mdc-button--raised": undefined,
          "mdc-button--unelevated": undefined,
        },
      ]
    }
  >
    test
  </button>
</Base>
`;

exports[`Button Tests Icon 1`] = `
<Base
  icon={true}
>
  <button
    className={
      Array [
        "mdc-icon-button",
        Object {
          "material-icons": true,
          "mdc-button--dense": undefined,
          "mdc-button--outlined": undefined,
          "mdc-button--raised": undefined,
          "mdc-button--unelevated": undefined,
        },
      ]
    }
  >
    test
  </button>
</Base>
`;

exports[`Button Tests Outlined 1`] = `
<Base
  outlined={true}
>
  <button
    className={
      Array [
        "mdc-button",
        Object {
          "material-icons": undefined,
          "mdc-button--dense": undefined,
          "mdc-button--outlined": true,
          "mdc-button--raised": undefined,
          "mdc-button--unelevated": undefined,
        },
      ]
    }
  >
    test
  </button>
</Base>
`;

exports[`Button Tests Raised 1`] = `
<Base
  raised={true}
>
  <button
    className={
      Array [
        "mdc-button",
        Object {
          "material-icons": undefined,
          "mdc-button--dense": undefined,
          "mdc-button--outlined": undefined,
          "mdc-button--raised": true,
          "mdc-button--unelevated": undefined,
        },
      ]
    }
  >
    test
  </button>
</Base>
`;

exports[`Button Tests Unelevated 1`] = `
<Base
  unelevated={true}
>
  <button
    className={
      Array [
        "mdc-button",
        Object {
          "material-icons": undefined,
          "mdc-button--dense": undefined,
          "mdc-button--outlined": undefined,
          "mdc-button--raised": undefined,
          "mdc-button--unelevated": true,
        },
      ]
    }
  >
    test
  </button>
</Base>
`;
